### PR TITLE
feat(custom metrics url): add custom metrics url

### PR DESCRIPTION
**Pull request by eranamrani**
- adds "/metrics" path to default options
- add metricsPath param to custom options to allow changing default metrics path

### CODE REVIEW

**Comment by qlikossbuild** [comment]
Hi, and thanks for contributing!

Note: If you are part of Qlik R&D, you can apply to the `open-source` team in our _internal_ GitHub organization.

Before we can handle your pull request we need you to sign our contributor license agreement.
This is a one-time process and enables you to contribute without hassle to any of the qlik-oss repositories in the future.

And it only takes a few minutes! Here are some useful links to get you started:

* [Sign the CLA](https://github.com/qlik-oss/open-source/blob/master/sign-cla.md#sign-the-cla)
* [Rationale behind the CLA](https://github.com/qlik-oss/open-source/blob/master/cla-rationale.md)
* [Qlik R&D Open Source repository](https://github.com/qlik-oss/open-source#open-source-at-qlik-rd)
* [Official Slack team](http://branch.qlik.com/slack)

Having troubles? Feel free to reach out to us either here or on Slack!

**Note:** Once the CLA is sorted, you should rebase/touch this pull request to trigger an update of this CLA status.

**Comment by hairyhenderson** [comment]
Hi @eranamrani - thanks for the contribution. To sign the CLA, you need to submit it to the https://github.com/qlik-oss/open-source repository, not this one.

File: index.js [hunk 1]
```diff
@@ -24,6 +25,7 @@ class MetricsMiddleware {
   /**
    * @typedef MetricsOptions
    * @type {object}
+   * @property {string} metricsUrl - defines custom metrics url
```
Comment (hairyhenderson) [inline index.js:28]:
```suggestion
   * @property {string} metricsPath - defines custom metrics path
```

File: index.js [hunk 2]
```diff
@@ -112,7 +114,7 @@ class MetricsMiddleware {
     }
     promClient.collectDefaultMetrics();
 
-    this.router.get('/metrics', this.metricsRoute.bind(this));
+    this.router.get(this.options.metricsUrl, this.metricsRoute.bind(this));
```
Comment (hairyhenderson) [inline index.js:117]:
```suggestion
    this.router.get(this.options.metricsPath, this.metricsRoute.bind(this));
```

**Comment by hairyhenderson** [comment]
Thanks for updating @eranamrani. Please remove the CLA file from this PR and submit it to the correct repo first, and then I can merge this.

**Comment by hairyhenderson** [comment]
(don't worry about the CircleCI build failure - that's unrelated to the code)

**Comment by eranamrani** [comment]
Hi @hairyhenderson, when is this expected to be merged?

**Comment by hairyhenderson** [comment]
@eranamrani you need to sign the CLA still - not in this repo, but in the  https://github.com/qlik-oss/open-source repo, as per the instructions (and as I mentioned 10 days ago: https://github.com/qlik-oss/http-metrics-middleware/pull/118#issuecomment-619592223)

**Comment by peol** [comment]
@hairyhenderson @eranamrani CLA approved: https://github.com/qlik-oss/open-source/pull/48

You need to remove the contributors file from this PR though.

**Comment by hairyhenderson** [comment]
I deleted the file 😉- will merge once CI completes!

**Pull request merged**